### ISSUE TITLE
AnimationTransitions: Quickly Change from `a` to `b`, then back to `a`will freeze the animation

### ISSUE THREAD

**Issue by xx1adfasd** [bevyengine/bevy#14548]
bevy 0.14

Quickly Change the animation from `a` to `b`, then back to `a`will stop the animation

The official example animation_fox is enough to show this issue.

Recreating the issue:

in line 176

```rust
*current_animation = (*current_animation + 1) % animations.animations.len();

transitions
    .play(
        &mut player,
        animations.animations[*current_animation],
        Duration::from_millis(250),
    )
    .repeat();
```
change it to 
```rust
*current_animation = (*current_animation + 1) % 2;

// the same
```
then run the example, quickly hit "Enter" multiple times, and you'll see the animation get stuck. 

Not just the example. In fact I got this issue in my own project, and I think it's a bevy bug, and then I triggered it easily in the fox example.

I think it's fatal, because in my situation, it's easily triggered when my PC is chaning its walking direction. Because there will be a moment the speed is zero. So, speed 1, 0, -1 will correspond to Walk, idle, Walk, and this is exactly the case.

**Comment by xx1adfasd** [comment]
I think it's because the `expire_completed_transitions` system.

it calls `player.stop(transition.animation)` which removes all the animations with the NodeIndex `transition.animation`,

but in this case, 

```
a
b
a'
```

the a' is different than a, so we can't use this methodology.

**Comment by alice-i-cecile** [comment]
Pretty confident this is a duplicate of https://github.com/bevyengine/bevy/issues/13910. Please take a look on main, and if not's already fixed report back and re-open this issue :)

**Comment by xx1adfasd** [comment]
You are right. Exactly the same issue. Apparently I didn't check the closed issues. But forgive me, there's tons of them. Just glad that this is fixed ^_^
